elasticsearch_dsl/faceted_search_base.py

```diff
@@ -316,36 +337,44 @@ def search(self):
 
     index = None
     doc_types = None
-    fields = None
-    facets = {}
+    fields: List[str] = []
+    facets: Dict[str, Facet[_R]] = {}
     using = "default"
 
-    def __init__(self, query=None, filters={}, sort=()):
+    def __init__(
+        self,
+        query: Optional[Query] = None,
+        filters: Dict[str, FilterValueType] = {},
+        sort: List[str] = [],
+    ):
         """
         :arg query: the text to search for
         :arg filters: facet values to filter
         :arg sort: sort information to be passed to :class:`~elasticsearch_dsl.Search`
         """
         self._query = query
-        self._filters = {}
+        self._filters: Dict[str, Query] = {}
         self._sort = sort
-        self.filter_values = {}
+        self.filter_values: Dict[str, List[FilterValueType]] = {}
         for name, value in filters.items():
             self.add_filter(name, value)
 
         self._s = self.build_search()
 
-    def count(self):
+    def count(self) -> int:
         return self._s.count()
 
-    def __getitem__(self, k):
+    def __getitem__(self, k: Union[int, slice]) -> "FacetedSearchBase[_R]":
         self._s = self._s[k]
         return self
 
-    def __iter__(self):
-        return iter(self._s)
+    # TODO this goes in async/sync subclasses!
+    # def __iter__(self) -> Iterator[Response[_R]]:
+    #    return iter(self._s)
```

Comment:
Did you intend to leave this TODO?

Reply:
No, that is code that needs to be deleted. Thanks!